contracts/base/snx-base/SNXRewardUniLPStrategyWithBuyback.sol

```diff
@@ -0,0 +1,325 @@
+
+      // sell reward token to token1
+      // we can accept 1 as minimum because this is called only by a trusted role
+
+      uint256 token0Amount;
+
+      if (uniswapRoutes[uniLPComponentToken0].length > 1) {
+        // in some cases, the reward token is the same as one of the components
+        // only swap when this is NOT the case
+
+        IUniswapV2Router02(uniswapRouterV2).swapExactTokensForTokens(
+          remainingRewardBalance/2,
+          amountOutMin,
+          uniswapRoutes[address(uniLPComponentToken0)],
+          address(this),
+          block.timestamp
+        );
+
+        token0Amount = IERC20(uniLPComponentToken0).balanceOf(address(this));
+        remainingRewardBalance = IERC20(rewardToken).balanceOf(address(this));
+      } else {
+        // no swap, just adjust the numbers
+        token0Amount = remainingRewardBalance/2;
+        remainingRewardBalance = remainingRewardBalance.sub(token0Amount);
+      }
+
+      // sell reward token to token2
+      // we can accept 1 as minimum because this is called only by a trusted role
+
+      if (uniswapRoutes[uniLPComponentToken1].length > 1) {
+        // in some cases, the reward token is the same as one of the components
+        // only swap when this is NOT the case
+        IUniswapV2Router02(uniswapRouterV2).swapExactTokensForTokens(
+          remainingRewardBalance,
+          amountOutMin,
+          uniswapRoutes[uniLPComponentToken1],
+          address(this),
+          block.timestamp
+        );
+      }
+      uint256 token1Amount = IERC20(uniLPComponentToken1).balanceOf(address(this));
+
+      // provide token1 and token2 to UniLPToken
+
+      IERC20(uniLPComponentToken0).safeApprove(uniswapRouterV2, 0);
+      IERC20(uniLPComponentToken0).safeApprove(uniswapRouterV2, token0Amount);
+
+      IERC20(uniLPComponentToken1).safeApprove(uniswapRouterV2, 0);
+      IERC20(uniLPComponentToken1).safeApprove(uniswapRouterV2, token1Amount);
+
+      uint256 liquidity;
+      (,,liquidity) = IUniswapV2Router02(uniswapRouterV2).addLiquidity(
+        uniLPComponentToken0,
+        uniLPComponentToken1,
+        token0Amount,
+        token1Amount,
+        1,  // we are willing to take whatever the pair gives us
+        1,
+        address(this),
+        block.timestamp
+      );
+    }
+  }
+
+  /*
+  *   Stakes everything the strategy holds into the reward pool
+  */
+  function investAllUnderlying() internal onlyNotPausedInvesting {
+    // this check is needed, because most of the SNX reward pools will revert if
+    // you try to stake(0).
+    if(IERC20(underlying).balanceOf(address(this)) > 0) {
+      IERC20(underlying).approve(address(rewardPool), IERC20(underlying).balanceOf(address(this)));
+      rewardPool.stake(IERC20(underlying).balanceOf(address(this)));
+    }
+  }
+
+  /*
+  *   Withdraws all the asset to the vault
+  */
+  function withdrawAllToVault() public restricted {
+    if (address(rewardPool) != address(0)) {
```

Comment:
Probably don't need this check - we're setting `rewardPool` at the beginning, so it won't be zero.

Reply:
Removed